src/main/java/org/jlab/clas/std/services/convertors/EtRingToHipoReader.java

```diff
@@ -0,0 +1,293 @@
+    }
+
+
+    public static class EtReader implements Closeable {
+
+        private static final String DEFAULT_SYS_NAME = "clara_system";
+        private static final String DEFAULT_STAT_NAME = "clara_station";
+
+        private static final Integer MAX_NEVENTS = 10;
+        private static final Integer TIMEOUT = 2000000;
+        private static final Integer QSIZE = 10;
+        private int runNumber;
+        private float torusField;
+        private float solenoidField;
+
+        private final EtSystem sys;
+        private final EtAttachment att;
+
+        private final CLASDecoder decoder = new CLASDecoder();
+        private EvioDataDictionary dict = null;
+        private final Queue<HipoEvent> evQueue = new LinkedList<>();
+
+        EtReader(String system, String host, int port, int runNumber,
+                 float torusField, float solenoidField) throws Exception {
+            this.runNumber = runNumber;
+            this.torusField = torusField;
+            this.solenoidField = solenoidField;
+            EtSystemOpenConfig config = new EtSystemOpenConfig(system, host, port);
+            config.setConnectRemotely(true);
+            sys = new EtSystem(config, EtConstants.debugInfo);
+            sys.open();
+
+            EtStationConfig statConfig = new EtStationConfig();
+            statConfig.setFlowMode(EtConstants.stationSerial);
+            statConfig.setBlockMode(EtConstants.stationNonBlocking);
+            statConfig.setCue(QSIZE);
+            dict = new EvioDataDictionary();
+            dict.initWithEnv("CLAS12DIR", "/etc/bankdefs/clas12");
+            EtStation stat = sys.createStation(statConfig, DEFAULT_STAT_NAME);
+            att = sys.attach(stat);
+
+            System.out.println("INFO: Start processing with runNumber = " + runNumber
+                + ", torus = " + this.torusField
+                + ", solenoid = " + this.solenoidField);
+        }
+
+        HipoEvent getEvent() throws Exception {
+            if (evQueue.isEmpty()) {
+                getEtEvents();
+            }
+            return evQueue.poll();
+        }
+
+        private void getEtEvents() throws Exception {
+
+            EtEvent[] mevs;
+            try {
+
+
+                // get events from ET system
+                try {
+                    mevs = sys.getEvents(att, Mode.SLEEP, null, TIMEOUT, MAX_NEVENTS);
+//                    mevs = sys.getEvents(att, Mode.TIMED, null, TIMEOUT, MAX_NEVENTS);
+                } catch (EtTimeoutException e) {
+                    System.out.println("Timed out, try again");
+                    return;
+                }
+
+                // ----------- ET
+                for (EtEvent mev : mevs) {
+                    // Get event's data buffer
+                    // buf.limit() = length of the actual data (not buffer capacity)
+                    ByteBuffer buf = mev.getDataBuffer();
+                    buf.limit(mev.getLength()).position(0);
```

Comment:
Why is this required? Doesn't the ET API return the buffer ready to use?

Reply:
I am not ET expert. I am just using suggestions from the main author here.